### PR TITLE
creating a guide for ERT configuration

### PR DESCRIPTION
**Pull request by mortalisk**
Have for now put a WIP version in file:///scratch/ert/mbend/docbuild/getting_started/configuration/poly_new/guide.html

### CODE REVIEW

**Comment by dotfloat** [comment]
I had a quick look-through. In `Increasing the number of realizations`:

`GEN_DATA` appears in the config suddenly, even though it has never been
mentioned before, `GEN_KW` was explained in the previous _Configuring the
parameter set and and corresponding template_ section, but not `GEN_DATA`.

Additionally, I think it'd be nice to have the keywords link to the reference. So, for example, clicking on `RUNPATH` will take you to the correct point in _List of Keywords_ page. Seems like you can do this relatively easily with Sphinx (cross-referencing). Although this may just be postponed for a later Issue.

**Comment by mortalisk** [comment]
I have addressed the issues you mentioned @dotfloat I simply removed the GEN_DATA, as it was not supposed to be introduced until the next chapter

File: test-data/local/poly_example/poly.ert [hunk 1]
```diff
@@ -16,3 +16,5 @@ GEN_DATA POLY_RES RESULT_FILE:poly_%d.out REPORT_STEPS:0 INPUT_FORMAT:ASCII
 
 INSTALL_JOB poly_eval POLY_EVAL
 SIMULATION_JOB poly_eval
+
+--hei
```
Comment (oyvindeide) [inline test-data/local/poly_example/poly.ert]: Guessing this isn't meant to be here?

**Comment by oyvindeide** [comment]
Is this ready for review? Can you squash the commits and upload the docs so it's easier to comment?

**Comment by mortalisk** [comment]
I squashed the commits. The docs are uploaded at file:///scratch/ert/mbend/docbuild/getting_started/configuration/poly_new/guide.html

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 2]
```diff
@@ -0,0 +1,291 @@
+Creating a config
+=================
+Ert runs a set of simulations with prior set of parameters sampled from distributions. The results of the simulations are then given to an optimization algorithm which uses a set of observations to tune the parameters in order to make a next run of the simulations produce results that fit better with the set of observations.
+
+To understand the basics of how to configure and use ERT, we will walk through how to set up a simple project that uses a simple polynomial evaluation as the forward model.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Would remove one (or both) `simple` from this sentence.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 3]
```diff
@@ -0,0 +1,291 @@
+Creating a config
+=================
+Ert runs a set of simulations with prior set of parameters sampled from distributions. The results of the simulations are then given to an optimization algorithm which uses a set of observations to tune the parameters in order to make a next run of the simulations produce results that fit better with the set of observations.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Stick to `ERT`

Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Think this sentence is a bit confusing, should maybe be expanded a bit or at least split in two.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+To understand the basics of how to configure and use ERT, we will walk through how to set up a simple project that uses a simple polynomial evaluation as the forward model.
+
+Minimal config
+--------------
+In order to test that our setup is working, we will create the minimal configuration we need in order to lauch the Graphical user interface and run an empty experiment that does nothing. First create a folder with the name "poly_example".
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Would remove the second `in order` to avoid too much repetition.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+In order to test that our setup is working, we will create the minimal configuration we need in order to lauch the Graphical user interface and run an empty experiment that does nothing. First create a folder with the name "poly_example". 
+
+Create a config
+***************
+Then create a file "poly.ert" inside the folder with the following content:
+
+    .. literalinclude:: minimal/poly.ert
+
+* :ref:`JOBNAME <jobname>` Specifies the name for each simulation. The %d will be replaced with the simulation number, so that the jobs will be called poly_0, poly_1, etc.
+* :ref:`NUM_REALIZATIONS <num_realizations>` We have to specify how many simulation we want. The word realization relates to the mathematical concept of an actual observation from a distribution.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Missing a `s` in simulations. Would reword `The word realization relates to the mathematical concept of an actual observation from a distribution.`,  as a realization is one member of an ensamble, so realizations sum up to ensamble size.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+* :ref:`QUEUE_SYSTEM <QUEUE_SYSTEM>` Finally, we must specify where each simulation is to run, which is handled by the queue system. LOCAL means that everything will run on you local computer. Other possibilites are LSF and TORQUE, which are queuing systems. To use these you must have an environment available where this is supported and configured.
+
+Launch the user interface
+*************************
+
+Navigate to this folder on your command line and run ERT::
+
+    ert gui poly.ert
+
+Wait a while and ERT will launch. You will see a warning because we haven't configured any observarions yet. This is expected, so click OK. You will see the main ERT user inteface screen.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: `Wait a while and ERT will launch` this will hopefully be improved and shouldn't be necessary to write in the guide?

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+
+.. image:: minimal/ert.png
+
+On the main screen you can see some elements. The ones we care about in this Guide are:
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: you `will`

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+
+* Simulation mode: lets you choose between running a single realization, or running all or specify a subset of them.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: `lets you choose between running a single realization, running all or specify a subset of them` which subset? Lets you select which algorithm to run (or ensemble experiment/single realization).

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+* Runpath: shows wehere each realization of the experiment will be executed. The "%d" in this path is replaced by the number of the realization. Here we see a default used, since we have not configured the runpath yet.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Typo `wehere` Would write `Here a default is used`, but might be personal preference.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+* The Configuration summary will show us what we have configured, and is empty at this moment.
+* The Start Simulation button will start an experiment with the current configuration and simulation mode.
+
+Run an empty experiment
+***********************
+To run an empty experiment, select "Ensamble experiment" as the Simulation mode. Then press the "Start simulation" button. You get a dialog asking if you want to start with the "default" case. That is fine, so press "Yes". The experiment will run, but since we have configured no forward model, only an empty experiment is executed.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: `"Start Simulation"`

Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Would write: `since we haven't configured a forward model`

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+
+A new window will appear called "Simulations". Here you can see the status of the running of the experiment. If you press "Details", you can see status of each running realization.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Would write: `Here you can see the status of the experiment`

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 12]
```diff
@@ -0,0 +1,291 @@
...
+----------------------------------
+The example project will use a simple polynomial evaluation as the forward model. In real experiments this would usually be running a simulator, like Eclipe, instead. The forward model is implemented as a simple python script "poly_eval.py". We will start with a simple script just to check that we can run it. Later we will expand on it and the configuration file to introduce some fundamental features of ERT.
+
+Add a Runpath
+*************
+In the minimal config, we used the default :ref:`RUNPATH <runpath>` which ran the realization in simulations/realization%d. This is not sufficient for running the algorithms of ERT. When running the algorithms of ERT, we must usually run the Ensamble several times, and the runpath needs to reflect this. We need to have each iteration run in a different forder, so that they won't step on each others toes. Add to the config ("poly.ert") file the following line::
+
+    RUNPATH poly_out/real_%d/iter_%d
+
+The second "%d" in the runpath will be replaced by the iteration of running the ensamble. In this way we can run several times, and have the results in different folders.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Is this true? Might be me misunderstanding what you're trying to convey. Running several times is handled by the case name, the iterations is for the iterative algorithms.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+The definition of a job is written in a separate file. In our case we only need to specify the executable name. Create a file called "POLY_EVAL" with the following content:
+
+    .. include:: with_simple_script/POLY_EVAL
+
+Now we must refer to this job definition in the configuration. Add the line::
+
+    INSTALL_JOB poly_eval POLY_EVAL 
+    SIMULATION_JOB poly_eval
+
+The :ref:`INSTALL_JOB <install_job>` line tells ERT the name of the job "poly_eval", and then the file in which to find the details of how to run the job. The :ref:`SIMULATION_JOB <simulation_job>` line tells ERT to run the job as part of the simulation run.
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Would change ` file in which to find` to: `file to` or possibly `file where to`

**Comment by sondreso** [comment]
I am moving this to todo considering the number of comments :)

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst (hunk 5 above)
Comment (mortalisk) [reply docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Removed the sentence and added some more info in the first paragraph

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst (hunk 12 above)
Comment (mortalisk) [reply docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: For some definitions of all the words, it is true. Alternatively for some definition of true.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 14]
```diff
@@ -0,0 +1,459 @@
...
+In this file we list each parameter line by line. The first part of a line is
+the name of the parameter. Following this is the type of distribution we want
+to sample the parameter from. Here we choose a uniform distribution. Following
+the distribution type, are the arguments describing the distribution. In the
+case of ``UNIFORM`` there are two parameters denoting the lower and upper bound
+of the distribution. Other distributions have different arguments.
+
+Adding a template
+*****************
+Then we create a template into which the samples from the distributions will be
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Very hard to read sentance

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 15]
```diff
@@ -0,0 +1,459 @@
...
+
+The arguments of ``GEN_DATA``:
+
+POLY_RES
+    Name of this result set.
+RESULT_FILE:poly_%d.out
+    File with results of simulation. The ``%d`` is always ``0``, but needs to
+    be specified. (it was used in earlier ERT versions)
+REPORT_STEPS:0
+    TODO
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Should there be a TODO here? If so the formatting should be different?

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 16]
```diff
@@ -0,0 +1,459 @@
...
+    .. literalinclude :: with_results/poly.ert
+
+If you now run the ensemble experiment again, and then open the plot view, you
+should see a new plot available called ``POLY_RES``:
+
+    .. image :: with_results/poly_plot.png
+
+Adding observations
+-------------------
+In order to use the update algorithms of ERT, we need to have some observations
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: `To use the update algorithms of ERT, we need to have some observations to compare with the results.` More to the point?

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 17]
```diff
@@ -0,0 +1,459 @@
...
+to which the results can be compared, so that ERT can tune the parameters to
+make the models better fit the observed data.
+
+The observations need to relate to some results of the simulation, so that the
+algorithms can compare them. We have some observations from the polynomial that
+were measured at the points 0, 2, 4, 6 and 8. The indices here happen to align
+with the x values of the polynomial evaluation, but this is incidental. Usually
+the indices are only the order of the observations in the file. Put the
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Clarify a bit? The indices are not only the order of the observations, but where they "match" the results indices. The text in index_list below is pretty clear:
```INDEX_LIST
    In our results file we have 10 values, while we only have 5 observation.
    This list tells ERT which of the results we have observations for. If they
    are the same length, you can omit this

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 18]
```diff
@@ -0,0 +1,459 @@
...
+following observations in the file ``poly_obs_data.txt``:
+
+    .. literalinclude:: with_observations/poly_obs_data.txt
+
+The observations are written one for each line, with the first number
+signifying the observed value, and the second number signifying the uncertainty
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Missing period.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst [hunk 19]
```diff
@@ -0,0 +1,459 @@
...
+
+Now lets describe the observations we have to ERT. This is done with the
+:ref:`OBS_CONFIG <obs_config>` keyword, which refers to a file in which we must
+describe the observations. Firs, make a file called ``observations`` in the
+project folder with the following content:
+
+    .. literalinclude:: with_observations/observations
+
+The ``GENERAL_OBSERVATION`` keyword tells ERT about how to find results, and how
```
Comment (oyvindeide) [inline docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Don't quite understand this sentence.

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst (hunk 15 above)
Comment (mortalisk) [reply docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: No, please inform we what makes sense to write there

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst (hunk 19 above)
Comment (mortalisk) [reply docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: Me neither. What would you write?

File: docs/rst/manual/getting_started/configuration/poly_new/guide.rst (hunk 15 above)
Comment (oyvindeide) [reply docs/rst/manual/getting_started/configuration/poly_new/guide.rst]: From the manual: 
```The REPORT_STEPS:0 is tightly bound to the %d integer format specifier in the result file - at load time the %d is replaced with the integer values given in the REPORT_STEPS: option, for the example given above that means that %d will be replaced with 0 and ERT will look for the file SimulatedWOC_0.txt. In principle it is possible to configure several report steps like: REPORT_STEPS:0,10,20 - then ERT will look for all three files SimulatedWOC_0.txt, SimultedWOC_10.txt and SimulatedWOC_20.txt. It is quite challenging to get this right, and the recommendation is to just stick with one result file at report step 0 3, in the future the possibility to load one keyword GEN_DATA for multiple report steps will probably be removed, but for now the GEN_DATA configuration is quite strict - it will fail if the RESULT_FILE attribute does not contain a %d.```
Think I would write something like: Tied to time map, for most cases this will be 0.
